File: order-model/src/main/scala/ordermodel/service/OrderApiService.scala
package ordermodel.service

import ordermodel.domain.Order
import ordermodel.usecases.PlaceNewOrderUsecase

import scala.beans.BeanProperty


/**
  * @author Bhuwan Upadhyay
  *
  */
class OrderApiService {

  @BeanProperty
  var placeNewOrderUsecase: PlaceNewOrderUsecase = _

  def placeNewOrder(newOrder: Order): List[String] = {
    placeNewOrderUsecase.newOrder(newOrder)
  }

  def listOrders(): List[Order] = {
    placeNewOrderUsecase.orderRepository.list()
  }

}

File: order-model/src/main/java/ordermodel/service/OrderApiService.java
package ordermodel.service;

import lombok.Setter;
import ordermodel.domain.Order;
import ordermodel.repository.OrderRepository;
import ordermodel.usecases.PlaceNewOrderUsecase;

import java.util.List;


/**
 * @author Bhuwan Upadhyay
 */
public class OrderApiService {

    @Setter
    private PlaceNewOrderUsecase placeNewOrderUsecase;
    @Setter
    private OrderRepository orderRepository;

    public List<String> placeNewOrder(Order newOrder) {
        return placeNewOrderUsecase.newOrder(newOrder);
    }

    public List<Order> listOrders() {
        return orderRepository.list();
    }

    public Order findOrder(String orderId) {
        return orderRepository.findById(orderId);
    }

    public Order saveOrder(Order order){
        return order;
    }
    public Order terminate(Order order){
        return order;
    }
}
